com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/ProcessConsole.java

```diff
@@ -17,19 +17,27 @@
 import java.io.InputStreamReader;
 import java.nio.charset.Charset;
 import java.nio.charset.StandardCharsets;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
 
+import io.reactivex.disposables.Disposable;
 import io.reactivex.functions.Consumer;
+import io.reactivex.schedulers.Schedulers;
 import io.reactivex.subjects.PublishSubject;
 
 public class ProcessConsole {
     private Process process;
     private String name;
     private Charset encoding;
-    private PublishSubject<String> stdoutSubject = PublishSubject.<String>create();
-    private PublishSubject<String> stderrSubject = PublishSubject.<String>create();
+    private PublishSubject<Message> outputSubject = PublishSubject.<Message>create();
     private Thread stdoutThread = null;
     private Thread stderrThread = null;
+    private AtomicInteger exit = new AtomicInteger(0);
+    private Disposable consumer = null;
 
+    /**
+     * constructor.
+     */
     public ProcessConsole(Process process) {
```

Comment:
OK. I got the whole point now. You are merging two streams. This can be done by simply calling Observable.mergeWith. It's the consumers' choice to merge those streams together. The underlying provider should keep it clean and simple.